timecache.go

```diff
@@ -0,0 +1,51 @@
+package ipfscluster
+
+import (
+	"time"
+
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+// TimeCache is an lru cache where entries are valid upto a specified
+// duration.
+type TimeCache struct {
```

Comment:
I have not bothered to add a mechanism to clean this cache at some interval. Worth doing so, may be for large clusters?

Reply:
Use https://github.com/whyrusleeping/timecache/

Reply:
This is taken from `whyrusleeping/timecache` and modified for our requirements.

Reply:
I don't see any similarities between this and whyrusleeping/timecache. It's worse by forgetting to delete things when they expire. Plus there is no need to modify anything for our requirements. Note the fact that one method was not authorized implies any other method will also not be authorized for the kind of rpc requests we're dealing with.

Reply:
By ` kind of rpc requests` you mean service name?

Reply:
I mean you are using service name for the timecache and that's not necessary.

Reply:
Now it uses whyrusleeping/timecache directly. Uses only peer id for timecache( if one request is not allowed, all wil be considered not allowed.)